homeassistant/components/notify/__init__.py

```diff
@@ -66,27 +66,31 @@ def send_message(hass, message, title=None, data=None):
 
 def setup(hass, config):
     """Setup the notify services."""
-    success = False
-
     descriptions = load_yaml_config_file(
         os.path.join(os.path.dirname(__file__), 'services.yaml'))
 
     targets = {}
 
-    for platform, p_config in config_per_platform(config, DOMAIN):
+    def setup_notify_platform(platform, p_config=None, discovery_info=None):
+        """Set up a notify platform."""
+        if p_config is None:
+            p_config = {}
+        if discovery_info is not None:
+            p_config = discovery_info
```

Comment:
I think that this is the wrong approach. This way a notify platform is unable to distinguish between discovery data (which cannot be trusted) and config from the user. Especially dangerous for `notify.command_line` !